src/relayclient/GSNConfigurator.ts

```diff
@@ -98,20 +100,30 @@ export function getDependencies (config: GSNConfig, provider?: HttpProvider, ove
   const httpClient = overrideDependencies?.httpClient ?? new HttpClient(new HttpWrapper(), config)
   const pingFilter = overrideDependencies?.pingFilter ?? GasPricePingFilter
   const relayFilter = overrideDependencies?.relayFilter ?? EmptyFilter
-  const asyncApprovalData = overrideDependencies?.asyncApprovalData ?? EmptyApprovalData
+  const asyncPaymasterData = overrideDependencies?.asyncPaymasterData ?? EmptyPaymasterData
   const scoreCalculator = overrideDependencies?.scoreCalculator ?? DefaultRelayScore
   const knownRelaysManager = overrideDependencies?.knownRelaysManager ?? new KnownRelaysManager(contractInteractor, config, relayFilter)
   const transactionValidator = overrideDependencies?.transactionValidator ?? new RelayedTransactionValidator(contractInteractor, config)
-  return {
+
+  const ret = {
     httpClient,
     contractInteractor,
     knownRelaysManager,
     accountManager,
     transactionValidator,
     pingFilter,
     relayFilter,
-    asyncApprovalData,
+    asyncPaymasterData,
     scoreCalculator,
     config
   }
+
+  // sanity check: overrides must on contain unknown fields.
```

Comment:
1. Why not? It can not be accessed by our TypeScript code anyways
2. TypeScript compiler does run such checks for you

Reply:
1. typescript compiler missed it in my test
2. your clients might be using javascript, not typescript, and you hide them from a breaking change (they can set fields that silently gets completely ignored)
3. need to find a library that does runtime schema check, and sanitize input params. can't rely on transpiled language.

Reply:
1. Check your code, do not introduce "sanity checks" to the code you are working with
2. If clients are using JavaScript their lives will be filled with pain and misery and I won't take on myself a mission to make their lives any better.
3. Not as part of Paymaster Data PR, ok? On your free time.

Reply:
added it because *I* hit this issue. added from the egoistic attempt to make my life easier (true, in earlier format of this PR, where I changed the asyncApprovalData to something else.)
Still, the minimum support we can give users is that if they use our lib, and later we change the API, then at least we should trigger an error if we notice they use an old code (I'm not saying we should make the lib backward-compatible, which is something that sometimes is useful, but at least fail fast.)